### PR TITLE
feat: export project

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
For any project - add functionality to export it using the dropdown menu in the top bar (the More button).

image

## Linear Ticket
https://linear.app/autokitteh/issue/UI-731/add-export-project-feature

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/topbar/project/buttons.tsx [hunk 1]
```diff
@@ -158,6 +158,43 @@ export const ProjectTopbarButtons = () => {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, []);
 
+	const exportProject = useCallback(async () => {
+		const { data: akProjectArchiveZip, error } = await ProjectsService.export(projectId!);
+
+		if (error) {
+			return;
```
Comment (J1za) [inline src/components/organisms/topbar/project/buttons.tsx:165]: Plz add Toast

File: src/components/organisms/topbar/project/buttons.tsx [hunk 2]
```diff
@@ -158,6 +158,43 @@ export const ProjectTopbarButtons = () => {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, []);
 
+	const exportProject = useCallback(async () => {
+		const { data: akProjectArchiveZip, error } = await ProjectsService.export(projectId!);
+
```
Comment (J1za) [inline src/components/organisms/topbar/project/buttons.tsx:163]: add state isExporting and `Loader` for Button to show that something is happening

File: src/components/organisms/topbar/project/buttons.tsx [hunk 3]
```diff
@@ -208,15 +245,36 @@ export const ProjectTopbarButtons = () => {
...
-					</Button>
+					<>
+						<Button
+							ariaLabel={t("topbar.buttons.export")}
+							className="group h-8 px-4 text-white"
+							onClick={exportProject}
+							variant="outline"
+						>
+							<IconSvg
+								className="fill-white transition group-hover:stroke-green-200 group-active:stroke-green-800"
```
Comment (J1za) [inline src/components/organisms/topbar/project/buttons.tsx:256]: hover and active color not working, I think remove `stroke` from svg icon

File: src/components/organisms/topbar/project/buttons.tsx [hunk 4]
```diff
@@ -158,6 +158,43 @@ export const ProjectTopbarButtons = () => {
...
+				year: "numeric",
+				hour: "2-digit",
+				minute: "2-digit",
+				hour12: false,
+			})
+			.replace(/[/:]/g, "")
+			.replace(", ", "-");
+
+		const fileName = `ak-${project?.name}-${dateTime}-archive.zip`;
+		link.download = fileName;
```
Comment (J1za) [inline src/components/organisms/topbar/project/buttons.tsx:189]:
```suggestion
		const link = Object.assign(document.createElement("a"), {
			href: url,
			download: fileName,
		});
```

File: src/components/organisms/topbar/project/buttons.tsx [hunk 5]
```diff
@@ -158,6 +158,43 @@ export const ProjectTopbarButtons = () => {
...
+		const { data: akProjectArchiveZip, error } = await ProjectsService.export(projectId!);
+
+		if (error) {
+			return;
+		}
+
+		const blob = new Blob([akProjectArchiveZip!], { type: "application/zip" });
+		const url = URL.createObjectURL(blob);
+		const link = document.createElement("a");
+		link.href = url;
```
Comment (J1za) [inline src/components/organisms/topbar/project/buttons.tsx:171]: now can remove it

File: src/components/organisms/topbar/project/buttons.tsx (hunk 1 above)
Comment (RonenMars) [reply src/components/organisms/topbar/project/buttons.tsx:165]: done

File: src/components/organisms/topbar/project/buttons.tsx (hunk 2 above)
Comment (RonenMars) [reply src/components/organisms/topbar/project/buttons.tsx:163]: done

File: src/components/organisms/topbar/project/buttons.tsx (hunk 3 above)
Comment (RonenMars) [reply src/components/organisms/topbar/project/buttons.tsx:256]: changed to `fill`

File: src/components/organisms/topbar/project/buttons.tsx (hunk 4 above)
Comment (RonenMars) [reply src/components/organisms/topbar/project/buttons.tsx:189]: done

File: src/components/organisms/topbar/project/buttons.tsx (hunk 5 above)
Comment (RonenMars) [reply src/components/organisms/topbar/project/buttons.tsx:171]: done

**Review by J1za — APPROVED** [review]

**Pull request merged**